src/layers/core/line-layer/line-layer.js

```diff
@@ -49,15 +49,13 @@ export default class LineLayer extends Layer {
   }
 
   draw({uniforms}) {
-    const {gl} = this.context;
-    const lineWidth = this.screenToDevicePixels(this.props.strokeWidth);
-    gl.lineWidth(lineWidth);
-    this.state.model.render(uniforms);
-    // Setting line width back to 1 is here to workaround a Google Chrome bug
-    // gl.clear() and gl.isEnabled() will return GL_INVALID_VALUE even with
-    // correct parameter
-    // This is not happening on Safari and Firefox
-    gl.lineWidth(1.0);
+    const {strokeWidth} = this.props;
+    const {viewport: {width, height}} = this.context;
+
+    this.state.model.render(Object.assign({}, uniforms, {
+      screenSize: [width, height],
```

Comment:
will be applied to both screen size and stroke width, then cancel each other out
> Ahh, you are probably right. Still, I think this code risks being hard to maintain if we make too many clever assumptions like that - even if we document them.

Unless there is a real perf gain (which I don't think there is here), it is usually better to work with the quantities and coordinate system that one would intuitively expect. In this case if we have a `screenSize` uniform, we'd have it specified in the amount of pixels the shader actually sees. That would make the same uniform easily usable in other contexts.

In fact `screenSize` (in DPI adjusted pixels) might be a common uniform that the `project` shader package could prepare by default (alongside `projectionMatrix` etc). Maybe longer term, your pixel space calculation function could also be added to that package (if general enough).

Not necessarily required for this diff, just sharing my thoughts... @shaojingli